src/lib/array.jl

```diff
@@ -171,8 +171,12 @@ _backmean(xs, Δ, dims) = zero(xs) .+ Δ ./ mapreduce(i -> size(xs,i),*,dims)
   end
 end
 
-@adjoint transpose(x) = transpose(x), Δ -> (transpose(Δ),)
-@adjoint Base.adjoint(x) = x', Δ -> (Δ',)
+_transpose(x) = transpose(x)
+_transpose(x::NamedTuple{(:parent,),<:Any}) = x.parent
```

Comment:
This looks generally good, but I'd prefer `_transpose` to be written as an inner function `back`.

Reply:
moved backward-actions to inner functions in both `adjoint` and `transpose`